lib/bots/versionbot.ts

```diff
@@ -1044,14 +1043,15 @@ export class VersionBot extends ProcBot {
 		const base = data.pull_request.base;
 		const owner = base.repo.owner.login;
 		const repo = base.repo.name;
+		const labels: GithubApiTypes.Label[] = event.cookedEvent.labels;
 		const headRepoFullName = `${head.repo.owner.login}/${head.repo.name}`;
 		let newVersion: string;
 		let fullPath: string;
 		let branchName = pr.head.ref;
 		let botConfig: VersionBotConfiguration;
 
-		// Only carry out merging when the label has been applied.
-		if (cookedData.data.action !== 'labeled') {
+		// Only carry out merging when the label has been applied and it's not a WIP.
+		if ((cookedData.data.action !== 'labeled') || _.some(labels, (label) => label.name === WIPLabel)) {
```

Comment:
You have the same snippet 5/6 times, added just in this PR - it may be worth adding it to it's own function.